### PR TITLE
caf: Added 1.0.2. Modernized

### PR DESCRIPTION
**Pull request by franramirez688**
### Summary
Added:  **caf/1.0.2**

More:
* Removed the `test_v1_package` folder.
* Modernized the recipe.
* Removed older patches.

Closes: https://github.com/conan-io/conan-center-index/issues/26081

### CODE REVIEW

File: recipes/caf/all/conanfile.py [hunk 1]
```diff
@@ -31,21 +32,6 @@ class CAFConan(ConanFile):
...
-    @property
-    def _min_cppstd(self):
-        return "17"
-
-    @property
-    def _minimum_compilers_version(self):
-        return {
-            "Visual Studio": "16",
-            "msvc": "192",
-            "gcc": "7" if Version(self.version) < "1.0.0" else "8",
```
Comment (franramirez688) [inline recipes/caf/all/conanfile.py:43]: I did not see why this was set so.

File: recipes/caf/all/conanfile.py [hunk 2]
```diff
@@ -87,10 +69,11 @@ def source(self):
...
     def generate(self):
         tc = CMakeToolchain(self)
-        if not valid_min_cppstd(self, self._min_cppstd):
-            tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
+        cppstd = str(self.settings.compiler.cppstd).replace("gnu", "")
+        if Version(self.version) < "1.0.0":
+            tc.variables["CMAKE_CXX_STANDARD"] = cppstd
         else:
-            tc.variables["CAF_CXX_VERSION"] = str(self.settings.compiler.cppstd).replace("gnu", "")
+            tc.variables["CAF_CXX_VERSION"] = cppstd
```
Comment (AbrilRBS) [inline recipes/caf/all/conanfile.py:76]:
```
if Version(self.version) >= "1.0.0":
    tc.variables["CAF_CXX_VERSION"] = cppstd
```

**Comment by uilianries** [comment]
The linter is accusing about using .value from configOptions: https://github.com/conan-io/conan-center-index/pull/26084/checks?check_run_id=33770465159

It would be better using `tc.variables["CAF_LOG_LEVEL"] = str(self.options.log_level).upper()`

**Review by uilianries — APPROVED** [review]
LGTM. I just checked the log-level and it could not be moved to conan conf, as it's used directly into .cpp files and affect the package ID.

**Review by perseoGI — APPROVED** [review]
LGTM

**Pull request merged**